analytics/src/main/java/com/segment/analytics/Properties.java

```diff
@@ -422,7 +423,7 @@ public String name() {
         }
 
         public String id() {
-            return getString(ID_KEY);
+            return getString(ID_KEY) != null ? getString(ID_KEY) : getString(PRODUCT_ID_KEY);
```

Comment:
question: do u think we should give first preference to `product_id` and then fall back to `id` or vice versa?

Reply:
sounds good